### PR TITLE
Implement flickr tags in pages

### PR DESCRIPTION
**Pull request by Matael**
The proposed flickr tag was implemented for articles only. We implemented a generic replacer taking a type of content as a second parameter. Calls to replace_{article,page}_tags now call the generic replacer in order to do the job for pages too. Tested on pelican 3.5.0

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -54,7 +54,7 @@ Settings
 
 .. code-block:: html
 
-    <p class="caption-container">
+    <spam class="caption-container">
```
Comment (streeter) [inline README.rst]: Why did you change this to a `spam` block?

File: README.rst [hunk 2]
```diff
@@ -74,11 +74,14 @@ Settings
...
 
+``FLICKR_TAG_PLACE_HOLDER_PICT`` -  variable is mandatory in your config if you dont have ``FLICKR_API_KEY`` pointing to a local pict how serve as a placeholder
+
+``FLICKR_TAG_PLACE_HOLDER_LINK`` - link associated with placeholder pict Default to git repo of this plug-in
 
 Flickr Settings
 ---------------
 
-The following two settings are required. In order to set them up, you will need to set up a Flickr API key. You can do this by `creating an app on Flickr`_. If the blog is a personal blog, then apply for a non-commercial key. Once you've got your key and secret, add them to your `Pelican configuration`_.
+The following two settings are required. In order to set them up, you will need to set up a Flickr API key. You can do this by `creating an app on Flickr`_. If the blog is a personal blog, then apply for a non-commercial key. Once you've got your key and secret, add them apply `pelicanconf.py changes`_.
```
Comment (streeter) [inline README.rst]: Not sure of this change. Why did you change the text?

Comment (jblb) [inline README.rst]: the idea behind this is to exclude Flickr API key from pelicanconf.py file and then exclude the flickr_api.py file from version system like github.com

File: README.rst (hunk 1 above)
Comment (jblb) [inline README.rst]: probably a other typo.... span is not spam :(

File: pelican_flickrtag/plugin.py [hunk 3]
```diff
@@ -25,7 +25,7 @@
             {% endif %} />
     </a>
     <span class="caption-text muted">{{title}}</span>
-</p>"""
+</spam>"""
```
Comment (streeter) [inline pelican_flickrtag/plugin.py]: Should this also be `span`?

File: README.rst [hunk 4]
```diff
@@ -66,7 +66,7 @@ Settings
                 {% endif %} />
         </a>
         <span class="caption-text muted">{{title}}</span>
-    </p>
+    </spam>
```
Comment (streeter) [inline README.rst]: This should also be `span`, I believe.

File: pelican_flickrtag/plugin.py [hunk 5]
```diff
@@ -101,17 +119,25 @@ def replace_article_tags(generator):
...
+                    'url': photo.url,
+                }
+
+                if include_dimensions:
+                    sizes = photo.getSizes()
+                    size = size_for_alias(sizes, size_alias)
+                    photo_mapping[id]['width'] = size['width']
+                    photo_mapping[id]['height'] = size['height']
+
+            except:
```
Comment (streeter) [inline pelican_flickrtag/plugin.py]: I'd prefer to not catch all exceptions here, with a catch-all. What was the specific exception type?

**Comment by streeter** [comment]
I like the idea, though I'll probably try to clean it up a bit before I merge it in.

File: pelican_flickrtag/plugin.py (hunk 5 above)
Comment (jblb) [inline pelican_flickrtag/plugin.py]: pelican_flickrtag.flickr.FlickrError: ERROR [1]: Photo not found

Comment (jblb) [inline pelican_flickrtag/plugin.py]: In a collaborative way of works, not everyone got flickr_api. when we write pages if we dont get them a place holder pict is displayed. will probably get a better way to do this.

File: README.rst (hunk 2 above)
Comment (gpoo) [inline README.rst]: The change is unrelated to supporting tags in pages. It would be simpler to have a separate pull request.

File: README.rst [hunk 6]
```diff
@@ -92,6 +95,25 @@ A Flickr API token is only required if you want to access photos that are privat
...
+    FLICKR_USER = ''
+    FLICKR_TAG_CACHE_LOCATION = './tmp_flickr'
+    FLICKR_TAG_TEMPLATE_NAME = 'images'
+    FLICKR_TAG_PLACE_HOLDER_PICT = "/theme/images/place-holder.png"
+    try:
+	   from flickr_api import *
+    except ImportError:
+	   pass
+
+and create a file mamed flickr_api.py containing your key, secret and other settings
```
Comment (gpoo) [inline README.rst]: Ditto.

File: pelican_flickrtag/plugin.py [hunk 7]
```diff
@@ -41,7 +41,18 @@ def setup_flickr(generator):
...
+    except:
+        logger.error ('[flickrtag]: FLICKR_TAG_PLACE_HOLDER_PICT variable is mandatory in your config')
 
+    try:
+        place_holder_link = generator.settings['FLICKR_TAG_PLACE_HOLDER_LINK']
+    except:
+        generator.settings.setdefault('FLICKR_TAG_PLACE_HOLDER_LINK','https://github.com/haum/pelican-flickrtag')        
+        logger.warning ('[flickrtag]: FLICKR_TAG_PLACE_HOLDER_LINK is set to default')
+    
+
```
Comment (gpoo) [inline pelican_flickrtag/plugin.py]: Ditto. This is unrelated to add support for pages.

File: pelican_flickrtag/plugin.py [hunk 8]
```diff
@@ -145,11 +175,15 @@ def replace_article_tags(generator):
             # Render the template
             replacement = template.render(context)
 
-            article._content = article._content.replace(match[0], replacement)
+            item._content = item._content.replace(match[0], replacement)
+
+
+def replace_article_tags(generator):
+    generic_replace(generator, 'article')
```
Comment (gpoo) [inline pelican_flickrtag/plugin.py]: I think you can use `generic_replace(generator, generator.articles)`. By doing that, you can remove the conditions and simplify the code. Probably, the downside would the the lack of a custom message of the photos found per article and page separately.

**Comment by streeter** [comment]
@Matael, #8 should have fixed this. Let me know if it missed something.
